h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -1781,67 +1786,67 @@ private void compactMoveChunks(Iterable<Chunk> move) {
             // now re-use the empty space
             reuseSpace = true;
             for (Chunk c : move) {
-                moveChunk(c, false);
+                moveChunk(c, 0);
             }
 
             // update the metadata (within the file)
             commit();
```

Comment:
Here, maybe wrong. For example, after the above movement, the right of file has been freed, and not persisted yet, but  the committing chunk(likely to get bigger than the freed one) may overwrite the right of the file!

Reply:
I did not get that. Chunks were just moved to the EOF, and that left enough free space within the file, so if worse come to worse, they now will fall back into same spaces (but more likely even close to the beginning of the file) and never overwrite their current location.

Reply:
@andreitokar The committing chunk may be bigger than the one before moving for new version getting bigger, and overwrite to the right freed(but not persisted) of the file. In addition, the committing chunk also may be partial write when power off, or the thread interrupted, then the store goes corrupt.

Reply:
@andreitokar The committing chunk may be bigger than the one before moving for version getting added, and overwrite to the right freed(but not persisted) of the file. In addition, the committing chunk also may be partial write when power off, or the thread interrupted, then the store goes corrupt.

Reply:
Ohh, I got it now. You are talking about chunk, produced by commit(). Yes, it may overlap and partial write may be a problem here. Let me sleep on it :smile:, literally.

Reply:
I think, last commit is addressing that issue now